xla/ffi/api/api.h

```diff
@@ -1313,7 +1313,8 @@ inline std::ostream& operator<<(std::ostream& os, const XLA_FFI_AttrType type) {
       auto* scalar = reinterpret_cast<XLA_FFI_Scalar*>(attr);         \
       if (XLA_FFI_PREDICT_FALSE(scalar->dtype != TYPE)) {             \
         return diagnostic.Emit("Wrong scalar data type: expected ")   \
-               << TYPE << " but got " << scalar->dtype;               \
+               << XLA_FFI_Get_DataType_Name(TYPE) << " but got "      \
```

Comment:
Can you add `operator<<` overload to api.h and forward https://github.com/openxla/xla/blob/main/xla/ffi/api/ffi.h#L65-L102 to it instead, just to have only one ToString conversion

Reply:
Why forward btw and not just move the overload from `ffi.h` to `api.h`?

Reply:
Nevermind, the class DataType is only defined in `ffi.h`.

Reply:
Yes, it's an `xla::PrimitiveType` re-implementation for user that can't depend on xla

Reply:
>add operator<< overload to api.h and forward

Done, please take a look.